docs/static_analysis.md

```diff
@@ -1,33 +1,296 @@
+Solidity Static Analysis
+========================
 
-This section gives information about the last compilation. By default, a
-new analysis is run at each compilation.
+Static code analysis is a process to debug the code by examining it and without actually executing the code. 
 
-The analysis tab gives detailed information about the contract code. It
-can help you avoid code mistakes and to enforce best practices.
+`Solidity Static Analysis` plugin performs static analysis on Solidity smart contracts once they are compiled. It helps to check security vulnerabilities and bad development practices. This plugin comes with `Solidity` environment of Remix IDE. It can also be activated individually from `Plugin Manager`.
+
+How to use
+------------
+
+If you select this plugin, you will see a number of modules listed along with checkboxes, one `Auto run` checkbox and a `Run` button.
 
 ![](images/a-static-analysis.png)
 
-Here is the list of analyzers:
+By default, all modules are selected for analysis and a new analysis is performed at each compilation.
+
+One can select/deselect the modules under which contract should be analyzed and can run the analysis again for last compiled contract by clicking on `Run`.
+
+If you don't want to run analysis each time you compile a contract, just uncheck the checkbox near to `Auto run`.
+
+Analysis Modules
+-----------------
+Currently, with Remix IDE v0.10.1, there are 21 analysis modules listed under 4 categories. Categories are: `Security`, `Gas & Economy`, `ERC` & `Miscellaneous`.
+
+Here is the list of modules under each category along with the example code which **should be avoided or used very carefully while development**:
+
+### Category: Security
+-   **Transaction origin: 'tx.origin' is used**
+
+`tx.origin` is useful only in very exceptional cases. If you use it for authentication, you usually want to replace it by "msg.sender", because otherwise any contract you call can act on your behalf.
+
+_Example:_
+```
+require(tx.origin == owner);
+```
+
+-   **Check effects: Potential reentrancy bugs**
+
+Potential Violation of Checks-Effects-Interaction pattern can lead to re-entrancy vulnerability.
+
+_Example:_
+```
+// sending ether first
+msg.sender.transfer(amount);
+
+// updating state afterwards
+balances[msg.sender] -= amount;
+```
+
+-   **Inline assembly: Inline assembly used**
+
+Use of inline assembly is advised only in rare cases.
+
+_Example:_
+```
+assembly {
+            // retrieve the size of the code, this needs assembly
+            let size := extcodesize(_addr)
+}
+```
+-   **Block timestamp: Semantics maybe unclear**
+
+`now` does not mean current time. `now` is an alias for `block.timestamp`. `block.timestamp` can be influenced by miners to a certain degree, be careful.
+
+_Example:_
+```
+// using now for date comparison
+if(startDate > now)
+    isStarted = true;
+
+// using block.timestamp
+uint c = block.timestamp;
+```
+-   **Low level calls: Semantics maybe unclear**
```

Comment:
I'm not sure why this section title has "Semantics may be unclear" - the problems don't seem to be about Semantics.

Reply:
This comes from libs. Lets discuss and improve that